integration/spark/src/main/common/java/io/openlineage/spark/agent/lifecycle/plan/CreateTableCommandVisitor.java

```diff
@@ -0,0 +1,48 @@
+package io.openlineage.spark.agent.lifecycle.plan;
+
+import io.openlineage.client.OpenLineage;
+import io.openlineage.spark.agent.util.PlanUtils;
+import java.util.Collections;
+import java.util.List;
+import org.apache.spark.sql.SparkSession;
+import org.apache.spark.sql.catalyst.analysis.NoSuchDatabaseException;
+import org.apache.spark.sql.catalyst.analysis.NoSuchTableException;
+import org.apache.spark.sql.catalyst.catalog.CatalogTable;
+import org.apache.spark.sql.catalyst.plans.logical.LogicalPlan;
+import org.apache.spark.sql.execution.command.CreateTableCommand;
+
+/**
+ * {@link LogicalPlan} visitor that matches an {@link CreateTableCommand} and extracts the output
+ * {@link OpenLineage.Dataset} being written.
+ */
+public class CreateTableCommandVisitor
+    extends QueryPlanVisitor<CreateTableCommand, OpenLineage.Dataset> {
+
+  private final SparkSession sparkSession;
+
+  public CreateTableCommandVisitor(SparkSession sparkSession) {
+    this.sparkSession = sparkSession;
+  }
+
+  @Override
+  public List<OpenLineage.Dataset> apply(LogicalPlan x) {
+    CreateTableCommand command = (CreateTableCommand) x;
+    CatalogTable catalogTable = command.table();
+    CatalogTable existingCatalogTable;
+
+    try {
+      existingCatalogTable =
+          sparkSession.sessionState().catalog().getTableMetadata(catalogTable.identifier());
+    } catch (NoSuchTableException | NoSuchDatabaseException e) {
+      existingCatalogTable = null;
+    }
+
+    if (existingCatalogTable != null
+        && existingCatalogTable.schema().equals(catalogTable.schema())) {
+      return Collections.singletonList(PlanUtils.getDataset(existingCatalogTable));
+    } else {
+      // apply triggered before applying the change - do not send an event
+      return Collections.emptyList();
+    }
```

Comment:
Why only post if the table creation is successful? In Marquez, if the job run fails after a start event that included the `OutputDataset`, the dataset would be created, but the current version would be unset, which I think seems sensible. Any harm in posting the `OutputDataset` here?

Reply:
You're right. I tried to fetch this information from `sparkSession` which was not the best approach and switched to retrieve information from `CatalogTable`. Currently two events are sent which is included in integration tests.